firmware/shared/code/CAN.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * CAN.h
+ *
+ *  Created on: Aug 8, 2020
+ *      Author: Poornachander
+ */
+
+#ifndef SHARED_CODE_CAN_H_
+#define SHARED_CODE_CAN_H_
+
+#include "stm32f4xx.h"
+#include "protocol.h"
+
+typedef struct
+{
+	// GPIO
+	uint8_t 		rxPin;
+	uint8_t 		txPin;
+	GPIO_TypeDef *	GPIOPort;
+	uint8_t			AFNumber;
+
+    // CAN
+    CAN_TypeDef *   CANPeriph;
+
+    //Common
+	void 		(* enablePeripheralsClockCallback)(void);
+} CAN_HWConfig_S;
+
+typedef struct
+{
+    const CAN_HWConfig_S * const HWConfig;
+	void	(* messageReceivedCallback)(const protocol_MID_E messageID, const protocol_allMessages_U * const message);
+} CAN_config_S;
+
+void CAN_init(void);
+void CAN_run1ms(void);

```

Comment:
arent we running messages for times other than 1ms? Also we should ensure that messages sent on the 1ms callback have the lowest pritority. We dont want to starve the other messages.

Reply:
That should actually be removed. There is no function called `CAN_run1ms`.

Reply:
However, in CAN, the messages with the highest frequency will usually have the higher priority (lower message ID). Those messages usually contain more important messages.